ddln_imaginator.proto

```diff
@@ -21,6 +21,64 @@ enum ImageFormat {
     GRAY = 4;
 }
 
+message Environment {
+    message Clouds {
+        enum Type {
+            NONE = 0;
+            ALTOCUMULUS = 1;
+            CLOUDS_BASE = 2;
+            CUMULUS = 3;
+            NIMBOSTRATUS = 4;
+            STRATOCUMULUS = 5;
+            STRATUS = 6;
+        }
+        optional Type type = 1;
+        optional float altitude_m = 2;
+        optional Point3d velocity = 3; // Only use the directional x,y,z values to set the wind in the Engine.
+    }
+    optional Clouds clouds = 1;
+
+    message Haze {
+        enum Type {
+            DEFAULT = 0; // Visibility = 20000, density = 1
+            CUSTOM = 1; // Custom visibility and density as specified below
+        }
+        optional Type type = 1;
+        optional float visibility = 2; // recommended 0 - 20000
+        optional float density = 3; // recommended 0.001 - 100
+    }
+    optional Haze haze = 2;
+
+    message Fog {
+        optional bool enabled = 1;
+        optional float multiplier = 2 ; // recommended 0.01 - 0.0001
+    }
+    optional Fog fog = 3;
+
+    message Sun {
+        enum Type {
+            DEFAULT = 0;
+            CUSTOM = 1;
+        }
+        optional Type type = 1;
+        optional float azimuth_deg = 2;
+        optional float altitude_deg = 3; // should not exceed 90 degree.
+        optional float intensity = 4; // scaled 0 - 1
+    }
+    optional Sun sun = 4;
+
+    message Precipitation {
+        enum Type {
+            NONE = 0;
+            RAIN = 1;
+            SNOW = 2;
+        }
+        optional Type type = 1;
```

Comment:
can we have snow & rain at the same time?

Reply:
I don't see a use case for this.

Reply:
https://en.wikipedia.org/wiki/Rain_and_snow_mixed

Reply:
@choi0330: please contact Alex if he sees a need for this. Let's keep things simple wherever we can.

Reply:
Ok, I'll ask him

Reply:
Rain and snow will be separate.